### PR TITLE
Enable back deployment of Concurrency

### PR DESCRIPTION
**Pull request by sevki**
Enable back deployment of Vapor's concurrency APIs on older Apple platforms.

This also drops support for Swift versions < 5.5.2 to match SwiftNIO

### CODE REVIEW

**Comment by sevki** [comment]
ping @weissi

File: Package.swift [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.4
+// swift-tools-version: 5.5.2
```
Comment (weissi) [inline Package.swift:1]:
```suggestion
// swift-tools-version:5.5.2
```

File: Package.swift [hunk 2]
```diff
@@ -33,7 +33,7 @@ let package = Package(
         .package(url: "https://github.com/swift-server/swift-backtrace.git", from: "1.1.1"),
         
         // Event-driven network application framework for high performance protocol servers & clients, non-blocking.
-        .package(url: "https://github.com/apple/swift-nio.git", from: "2.42.0"),
+        .package(url: "https://github.com/apple/swift-nio.git", from: "2.44.0"),
```
Comment (0xTim) [inline Package.swift:36]: Could you also update async kit to 1.15.0?

File: Sources/Development/routes.swift [hunk 3]
```diff
@@ -282,7 +282,7 @@ public func routes(_ app: Application) throws {
         }
     }
     
-    @available(macOS 12, iOS 15, watchOS 8, tvOS 15, *)
+    @available(macOS 10.15, iOS 15, watchOS 8, tvOS 15, *)
```
Comment (0xTim) [inline Sources/Development/routes.swift:285]: We should enable back porting for all Apple Platforms

**Review by 0xTim — CHANGES_REQUESTED** [review]
Could you enable backporting for all Apple Platforms?

**Comment by sevki** [comment]
Happy new year 🥳 @0xTim! 

Would you mind taking another look?

**Comment by 0xTim** [comment]
Original comment: 

> preceded by https://github.com/vapor/async-kit/pull/97

**Review by 0xTim — APPROVED** [review]
Thanks!

**Pull request merged**